packages/blockchain-wallet-v4-frontend/src/data/modals/slice.ts

```diff
@@ -0,0 +1,64 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit'
+import { merge } from 'ramda'
+
+import {
+  ModalNameType,
+  ModalParamPropsType,
+  ModalsState,
+  UpdateModalOptionsPayload,
+  UpdateModalPayload
+} from './types'
+
+const initialState: ModalsState = []
+
+const modalsSlice = createSlice({
+  initialState,
+  name: 'modals',
+  reducers: {
+    closeAllModals: () => {
+      return initialState
+    },
+    closeModal: {
+      prepare: (modalName?: ModalNameType) => {
+        return { payload: { modalName } }
+      },
+      reducer: (state, action: PayloadAction<{ modalName?: ModalNameType }>) => {
+        if (action.payload.modalName) {
+          state = state.filter((modal) => modal.type !== action.payload.modalName)
```

Comment:
since the filter method in javascript returns a new array this will error. You need to `return` the new array. See here https://redux-toolkit.js.org/usage/immer-reducers#mutating-and-returning-state

Reply:
yeah immutable will complain, but fun fact it worked without warning :(